odoo/fields.py

```diff
@@ -2536,6 +2550,14 @@ def update_db(self, model, columns):
         return super(Many2one, self).update_db(model, columns)
 
     def update_db_column(self, model, column):
+        """ Advertize field column type as int8 if that's the current column type,
+        or if the referenced column is of type int8 """
+        comodel = model.env[self.comodel_name]
+        fk_type = sql.column_type(model.env.cr, comodel._table, 'id') or (
+            'int8' if comodel._bigint_id else 'int4')
```

Comment:
This is terrible from a performance point of view: every new many2one field makes a query (in `column_type`).  We currently have one query *per model*.  I have optimized this code a few months ago.

Reply:
Refactored by having `sql.table_columns` also fetch columns referenced from the requested table. That query has now become a lot heavier though.